plottr/plot/base.py

```diff
@@ -134,3 +144,315 @@ def makeFlowchartWithPlot(nodes: List[Tuple[str, Type[Node]]],
+            re_plotItem.data[-1] = re_data
+            re_plotItem.labels[-1] = re_label
+            im_plotItem.data[-1] = im_data
+            im_plotItem.labels[-1] = im_label
+            im_plotItem.id = re_plotItem.id + 1
+            return [re_plotItem, im_plotItem]
+
+        if self.complex_representation is ComplexRepresentation.magAndPhase:
+            mag_data = np.abs(plotItem.data[-1])
+            phase_data = np.angle(plotItem.data[-1])
+
+            if label == '':
+                mag_label, phase_label = 'Mag', 'Phase'
+            else:
+                mag_label, phase_label = label + ' (Mag)', label + ' (Phase)'
+
+            mag_plotItem = plotItem
+            phase_plotItem = deepcopy(mag_plotItem)
+
+            mag_plotItem.data[-1] = mag_data
+            mag_plotItem.labels[-1] = mag_label
+            phase_plotItem.data[-1] = phase_data
+            phase_plotItem.labels[-1] = phase_label
+            phase_plotItem.id = mag_plotItem.id + 1
+            phase_plotItem.subPlot = mag_plotItem.subPlot + 1
+            return [mag_plotItem, phase_plotItem]
+
+    # public methods
+    def nSubPlots(self):
+        ids = []
+        for id, item in self.plotItems.items():
+            ids.append(item.subPlot)
+        return len(set(ids))
+
+    def subPlotItems(self, subPlotId: int) -> OrderedDictType[int, PlotItem]:
+        items = OrderedDict()
+        for id, item in self.plotItems.items():
+            if item.subPlot == subPlotId:
+                items[id] = item
+        return items
+
+    def subPlotLabels(self, subPlotId: int) -> List[List[str]]:
+        ret = []
+        items = self.subPlotItems(subPlotId)
+        for id, item in items.items():
+            if item.labels is not None:
+                for i, l in enumerate(item.labels):
+                    while(len(ret)) <= i:
+                        ret.append([])
+                    ret[i].append(l)
+        return ret
+
+    def addData(self, *data: np.ndarray, join: Optional[int] = None,
+                labels: Optional[List[str]] = None, style: Optional[str] = None,
+                **plotOptions: Any) -> int:
+
+        id = _generate_auto_dict_key(self.plotItems)
+        if join is None:
+            subPlotId = self.nSubPlots()
+        else:
+            subPlotId = self.plotItems[join].subPlot
+
+        if labels is None:
+            labels = [''] * len(data)
+
+        plotItem = PlotItem(list(data), id, subPlotId,
+                            labels, style, plotOptions)
+
+        for p in self._splitComplexData(plotItem):
+            self.plotItems[p.id] = p
+
+        return id
+
+    # Methods to be implemented by inheriting classes
+    def makeAxes(self, nSubPlots: int) -> List[Any]:
+        raise NotImplementedError
+
+    def formatSubPlot(self, subPlotId: int):
+        pass
+
+    def plot_line(self, plotItem: PlotItem):
+        raise NotImplementedError
+
+    def plot_image(self, plotItem: PlotItem):
```

Comment:
shouldn't there be methods for each PlotType ?

Reply:
Good idea, this would probably be cleaner. See also comment above: i'd need to think a bit on how to make sure this remains extendable, and there still needs to be a way to translate this to multiple backends.